prov/src/test/java/org/bouncycastle/jce/provider/test/XIESTest.java

```diff
@@ -0,0 +1,307 @@
+        cipher.init(Cipher.ENCRYPT_MODE, keyPair.getPublic());
+
+        String toEncrypt = "Hello";
+
+        // Check that cipher works ok
+        cipher.doFinal(toEncrypt.getBytes());
+
+        // Using a SealedObject to encrypt the same string fails with a NullPointerException
+        SealedObject sealedObject = new SealedObject(toEncrypt, cipher);
+
+        cipher.init(Cipher.DECRYPT_MODE, keyPair.getPrivate());
+
+        String result = (String)sealedObject.getObject(cipher);
+
+        isTrue("result wrong", result.equals(toEncrypt));
+
+        result = (String)sealedObject.getObject(keyPair.getPrivate());
+
+        isTrue("result wrong", result.equals(toEncrypt));
+    }
+
+    public void doTest(
+        String              testname,
+        KeyPairGenerator    g,
+        String              cipher,
+        IESParameterSpec    p)
+        throws Exception
+    {
+
+        byte[] message = Hex.decode("0102030405060708090a0b0c0d0e0f10111213141516");
+        byte[] out1, out2;
+
+        // Generate static key pair
+        KeyPair KeyPair = g.generateKeyPair();
+        PublicKey Pub = KeyPair.getPublic();
+        PrivateKey  Priv = KeyPair.getPrivate();
+
+        Cipher c1 = Cipher.getInstance(cipher);
+        Cipher c2 = Cipher.getInstance(cipher);
+
+        // Testing with null parameters and DHAES mode off
+        c1.init(Cipher.ENCRYPT_MODE, Pub, new SecureRandom());
+        c2.init(Cipher.DECRYPT_MODE, Priv, c1.getParameters());
+
+        isTrue("nonce mismatch", Arrays.areEqual(c1.getIV(), c2.getIV()));
+
+        out1 = c1.doFinal(message, 0, message.length);
+        out2 = c2.doFinal(out1, 0, out1.length);
+        if (!areEqual(out2, message))
+        {
+            fail(testname + " test failed with null parameters, DHAES mode false.");
+        }
+
+        // Testing with given parameters and DHAES mode off
+        c1.init(Cipher.ENCRYPT_MODE, Pub, p, new SecureRandom());
+        c2.init(Cipher.DECRYPT_MODE, Priv, p);
+        out1 = c1.doFinal(message, 0, message.length);
+        out2 = c2.doFinal(out1, 0, out1.length);
+        if (!areEqual(out2, message))
+        {
+            fail(testname + " test failed with non-null parameters, DHAES mode false.");
+        }
+
+        //
+        // corrupted data test
+        //
+        int offset = out1.length - (message.length + 8);
+        byte[] tmp = new byte[out1.length];
+        for (int i = offset; i != out1.length; i++)
+        {
+            System.arraycopy(out1, 0, tmp, 0, tmp.length);
+            tmp[i] = (byte)~tmp[i];
+
+            try
+            {
+                c2.doFinal(tmp, 0, tmp.length);
+                fail("decrypted corrupted data");
+            }
+            catch (BadPaddingException e)
+            {
+                isTrue("wrong message: " + e.getMessage(), "unable to process block".equals(e.getMessage()));
+            }
+        }
+// TODO: DHAES mode is not currently implemented, perhaps it shouldn't be...
```

Comment:
Please remove the comments as it is not implemented as it may be merged. Create a separate issue to track that issue if it is useful to be implemented in the future

Reply:
Sorry I deleted the fork of bc-java, and cannot update the code based on this PR. Could you please delete these commented lines before/after the merge?

Reply:
I am not a committer, but just reading through the PR.